### PR TITLE
feat: use ConfigManager to validate the config

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**

### CODE REVIEW

**Review by cjihrig — APPROVED** [review]

**Comment by mcollina** [comment]
I think it would be better if this just bundled the generated JSON Schema instead. The action then could be thinner and not bring along all of Platformatic.

Wdyt @cjihrig @ivan-tymoshenko ?

**Comment by ivan-tymoshenko** [comment]
> I think it would be better if this just bundled the generated JSON Schema instead. The action then could be thinner and not bring along all of Platformatic.

Platformatic is the one who knows which formats the config can have, serialization formats (onestep should parse the config before validation), etc.

**Comment by mcollina** [comment]
This creates a tight coupling between onestep and platformatic. I would prefer to avoid that.

Maybe we can add a `platformatic db/service validate` command to validate the config and then defer to a version of platformatic installed in the action itself?

**Comment by ivan-tymoshenko** [comment]
So you will have to have a platformatic dependency in your package.json, if I understand you correctly?

**Comment by ivan-tymoshenko** [comment]
We also need to be able to parse a config.

**Comment by mcollina** [comment]
You'll still need to have a commit and a release done each time. It's two things to keep in sync... eventually they will get out of sync and create friction for our users.

**Comment by mcollina** [comment]
I think we should detect which version of Platformatic the project is using, validate against that, and send this information to the control-plane too.

File: index.js [hunk 1]
```diff
@@ -321,6 +303,86 @@ async function updatePlatformaticComment (octokit, commentId, comment) {
...
+  const localPackagePath = join(pathToProject, 'node_modules', packageName)
+  const localPackageInstalled = await isFileAccessible(localPackagePath)
+
+  if (localPackageInstalled) {
+    const localPackageVersion = await getPackageVersion(localPackagePath)
+    core.info(`Local version of ${packageName} is ${localPackageVersion}`)
+
+    if (localPackageVersion === latestPackageVersion) return localPackagePath
+  }
+
```
Comment (mcollina) [inline index.js:373]: If the local version is not the latest, let's print a warning:

"In our beta phase, we only support using the latest release of Platformatic".

Comment (ivan-tymoshenko) [reply index.js:373]: Should we throw if versions are not equal or warning would be enough?

Comment (mcollina) [reply index.js:377]: warning is enough for now

File: index.js [hunk 2]
```diff
@@ -321,6 +303,90 @@ async function updatePlatformaticComment (octokit, commentId, comment) {
...
+  const packageJsonExist = await isFileAccessible(packageJsonPath)
+  if (!packageJsonExist) return
+
+  const packageJsonData = await readFile(packageJsonPath, 'utf8')
+  const packageJson = JSON.parse(packageJsonData)
+
+  const dependencies = packageJson.dependencies
+  if (
+    dependencies !== undefined &&
+    dependencies.platformatic !== undefined
```
Comment (cjihrig) [inline index.js:322]: If optional chaining is always available this could be:

```js
if (packageJson?.dependencies?.platformatic) {
  ...
}
```

**Pull request closed without merging**